tests/phpunit/tests/html-api/README-performance.md

```diff
@@ -0,0 +1,153 @@
+
+### Benchmark Test Results
+Benchmark tests output throughput and memory information:
+```
+[BENCHMARK] Simple Selector Parsing: 67543.21 ops/sec, 0.0148ms avg, 0.00KB memory, 0.00KB peak
+```
+
+- **ops/sec**: Operations (parsings) per second
+- **avg**: Average time per operation in milliseconds
+- **memory**: Memory used during test in KB
+- **peak**: Peak memory usage in KB
+
+## Performance Expectations
+
+### Current Performance Thresholds
+- **Simple selectors**: > 50,000 ops/sec
+- **Compound selectors**: > 10,000 ops/sec
+- **Complex selectors**: > 5,000 ops/sec
+- **Selector lists**: > 3,000 ops/sec
+- **Attribute selectors**: > 8,000 ops/sec
+- **Unicode selectors**: > 15,000 ops/sec
+- **Error handling**: > 20,000 ops/sec
+
+### Memory Usage
+- Memory usage should scale roughly linearly with selector complexity
+- Error handling should not consume excessive memory
+- Memory leaks should be avoided during repeated parsing
+
+## Monitoring Performance
+
+### In CI/CD
+These tests can be integrated into CI/CD pipelines to:
+- Monitor performance regressions
+- Ensure performance standards are maintained
+- Compare performance across different PHP versions
+
+### Local Development
+Use these tests to:
+- Measure performance impact of code changes
+- Identify performance bottlenecks
+- Optimize parsing algorithms
+
+## Troubleshooting
+
+### Tests Failing Due to Performance
+If performance tests fail consistently:
+
+1. **Check system load**: High CPU usage can affect timing
+2. **Run multiple times**: Single runs may have variance
+3. **Update thresholds**: Hardware differences may require adjustment
+4. **Profile specific cases**: Use tools like Xdebug to identify bottlenecks
+
+### Adjusting Performance Thresholds
+Thresholds can be adjusted in the test files:
+- `wpCssSelectorPerformance.php`: Modify `PERFORMANCE_THRESHOLD_MS` constant
+- `wpCssSelectorBenchmark.php`: Modify assertion values in individual tests
+
+### Adding New Performance Tests
+When adding new selector functionality:
+1. Add parsing performance tests to `wpCssSelectorPerformance.php`
+2. Add throughput benchmarks to `wpCssSelectorBenchmark.php`
+3. Update this documentation with new performance expectations
+
+## Best Practices
+
+### Writing Performance Tests
+- Use realistic test data that represents actual usage
+- Include both positive and negative test cases
+- Test edge cases and error conditions
+- Consider memory usage alongside execution time
+
+### Interpreting Results
+- Look for trends over time, not just individual runs
+- Compare relative performance between different selector types
+- Consider the context of how selectors are used in real applications
+- Monitor both average and peak performance metrics
+
+### Optimization Guidelines
+Based on test results, focus optimization efforts on:
+- Most commonly used selector types
+- Cases with highest memory usage
+- Scenarios with slowest throughput
+- Error handling paths that are frequently exercised
```

Comment:
the tests are fine, but this long-winded and fluffy README is annoying me in its lack of substance. can we nix it completely, or start over with one that is information dense?